uPortal-soffit-renderer/src/main/java/org/apereo/portal/soffit/renderer/SoffitRendererController.java

```diff
@@ -106,72 +121,90 @@
     private String viewsLocation;
     private final Map<ViewTuple,String> availableViews = new HashMap<>();
 
+    private Map<AnnotatedElement,Object> modelAttributes;
+
     protected final Logger logger = LoggerFactory.getLogger(getClass());
 
+    @PostConstruct
+    public void init() {
+        /*
+         * Gather classes & methods that reference @SoffitMoldelAttribute
+         */
+        final Map<AnnotatedElement,Object> map = new HashMap<>();
+        final String[] beanNames = applicationContext.getBeanDefinitionNames();
+        for (String name : beanNames) {
```

Comment:
Would this be a place where [using Java 8 streams](https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html) would be effective?